pkg/cli/cvd.go

```diff
@@ -218,14 +235,25 @@ func (c *cvdCreator) createWithOpts() ([]*hoapi.CVD, error) {
 		AdditionalInstancesNum: c.opts.AdditionalInstancesNum(),
 	}
 	c.statePrinter.Print(stateMsgStartCVD)
-	res, err := c.service.HostService(c.opts.Host).CreateCVD(createReq, client.InjectedCredentials)
+	res, err := c.service.HostService(c.opts.Host).CreateCVD(createReq, c.credentialsFactory())
 	c.statePrinter.PrintDone(stateMsgStartCVD, err)
 	if err != nil {
 		return nil, err
 	}
 	return res.CVDs, nil
 }
 
+func credentialsFactoryFromSource(source string) (CredentialsFactory, error) {
+	switch source {
+	case NoneCredentialsSource:
+		return func() string { return "" }, nil
+	case InjectedCredentialsSource:
+		return func() string { return client.InjectedCredentials }, nil
+	default:
+		return nil, fmt.Errorf("unknown credentials source: %s", source)
```

Comment:
nit: would this be a good place to use `panic` instead?

Reply:
no: the library shouldn't panic, the executable binary that uses it can panic if it wants to, but not the library.
The rule of thumb is to panic only if the state of the system is corrupted, otherwise use regular error handling.